src/components/back-drop/back-drop.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import style from './back-drop.module.css'
+
+type BackdropProps = {
+  show: boolean
+  clicked: () => void
+  children: React.ReactNode
+}
+
+const Backdrop: React.FC<BackdropProps> = ({ show, clicked, children }) => {
+  return (
+    <>
+      {show ? <div className={style.Backdrop} onClick={clicked}></div> : null}
+      {children}
```

Comment:
Why the children are not in div?

Reply:
because if i place children inside it trnasition will not work properly, it will be state based